database/sqlserver/sqlserver.go

```diff
@@ -344,3 +364,21 @@ func (ss *SQLServer) ensureVersionTable() (err error) {
 
 	return nil
 }
+
+func getMSITokenProvider(resource string) (func() (string, error), error) {
+	msiEndpoint, err := adal.GetMSIEndpoint()
```

Comment:
double check the resource endpoint here isn't hardcoded as well

Reply:
can remove this call as it's deprecated: https://github.com/Azure/go-autorest/blob/autorest/adal/v0.9.14/autorest/adal/token.go#L736. adal.NewServicePrincipalTokenFromMSI covers this